### PR TITLE
AB2D 516 modify worker to skip benes who have opted out

### PR DESCRIPTION
**Pull request by spathiyil**
AB2D 516 modify worker to skip benes who have opted out

### Summary

AB2D 516 modify worker to skip benes who have opted out

### Checklist

- [x] Tests and linting pass

### Security
N/A

### Additional JIRA Tickets (optional)

N/A

### CODE REVIEW

File: worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java [hunk 1]
```diff
@@ -123,9 +126,18 @@ public Job processJob(final String jobId) {
...
+        var futureResourcesHandles = new ArrayList<Future<Integer>>();
+        for (PatientDTO patient : patients) {
+            var patientId = patient.getPatientId();
+
+            if (isOptOutPatient(patientId)) {
+                // This patient has opted out. Skip this patient and process the next record.
+                continue;
+            }
+
+            futureResourcesHandles.add(patientClaimsProcessor.process(patientId, lock, outputFile, errorFile));
```
Comment (dkrylovsb) [inline worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java]: You could probably use `filter` on that stream to keep the code neat

File: worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java [hunk 2]
```diff
@@ -146,6 +158,18 @@ public Job processJob(final String jobId) {
...
+    private boolean isOptOutPatient(String patientId) {
+
+        final List<Consent> consents = consentRepository.findByHicn(patientId);
+        if (consents.isEmpty()) {
+            // No opt-out record found for this patient - Opt-In by default.
+            return false;
+        }
+
+        // found at least one opt-out record for this patient. This patient has opted-out
+        return true;
```
Comment (dkrylovsb) [inline worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java]: Opt-out _could_ be effective on or after a certain date. So let's get the date from the `Consent` record, and if it is in the _future_, consider the Patient still _opted in_.

Comment (spathiyil) [reply worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java]: done

File: worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java (hunk 1 above)
Comment (spathiyil) [reply worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java]: done

**Pull request merged**